datadog-checks-base/datadog_checks/checks/prometheus/mixins.py

```diff
@@ -393,7 +394,14 @@ def process_metric(self, message, **kwargs):
                     self._submit(self.metrics_mapper[message.name], message, send_histograms_buckets, custom_tags)
                 except KeyError:
                     if not ignore_unmapped:
+                        # call magic method (non-generic check)
                         getattr(self, message.name)(message, **kwargs)
+                    else:
+                        # try matching wildcard (generic check)
+                        for metric in self.metrics_mapper.keys():
+                            if '*' in metric and fnmatchcase(message.name, metric):
```

Comment:
I'm wondering if it's worth it to create a sublist filtered with items matchin `'*' in metric` or if the operation is cheap enough